gnuradio-runtime/lib/pmt/pmt.cc

```diff
@@ -150,27 +150,11 @@ static std::vector<pmt_t>* get_symbol_hash_table()
 pmt_symbol::pmt_symbol(const std::string& name) : d_name(name) {}
 
 
-static unsigned int hash_string(const std::string& s)
-{
-    unsigned int h = 0;
-    unsigned int g = 0;
-
-    for (std::string::const_iterator p = s.begin(); p != s.end(); ++p) {
-        h = (h << 4) + (*p & 0xff);
-        g = h & 0xf0000000;
-        if (g) {
-            h = h ^ (g >> 24);
-            h = h ^ g;
-        }
-    }
-    return h;
-}
-
 bool is_symbol(const pmt_t& obj) { return obj->is_symbol(); }
 
 pmt_t string_to_symbol(const std::string& name)
 {
-    unsigned hash = hash_string(name) % get_symbol_hash_table_size();
+    unsigned hash = std::hash<std::string>()(name) % get_symbol_hash_table_size();
```

Comment:
Include what you use: according to all I know, `std::hash` is in `<functional>`, but we're including that nowhere; this still works (and `g++ -M -c -I/path/to/includes pmt.cc` tells me that, yes, somehow <functional> is recursively included), but we should really do `#include <functional>` at the top of this file.

Reply:
I had this originally but reading http://www.cplusplus.com/reference/functional/hash/ made me think that `std::hash<std::string>` lives in `<string>`. So I removed it and it still worked. I could be wrong.